packages/utils/src/media/CanvasRenderTarget.ts

```diff
@@ -10,10 +10,10 @@ import type { ICanvas, ICanvasRenderingContext2D } from '@pixi/settings';
 export class CanvasRenderTarget
 {
     /** The Canvas object that belongs to this CanvasRenderTarget. */
-    public canvas: ICanvas;
+    public canvas: ICanvas | null;
```

Comment:
We can use getter to expose a non-null `canvas`:
```ts
private _canvas: ICanvas | null;

/** The Canvas object that belongs to this CanvasRenderTarget. */
public get canvas(): ICanvas
{
    // #if _DEBUG
    this._checkDestroyed();
    // #endif

    return this._canvas;
}
```

And also for `context`. Internal use of `canvas` and `context` need to be change into `_canvas` and `_context` (and also in `asserts` of `_checkDestroyed`). After this we won't need `?.` in `renderTarget.canvas?.xxx`.

Reply:
I don't think `_checkDestroyed` is actually a type guard as typescript thinks the code can still reach the return value as it's not wrapped in an if statement or something.

```typescript
public get canvas(): ICanvas
{
    // #if _DEBUG
    this._checkDestroyed();
    // #endif

    return this._canvas;
}
 ```
 
This still returns  `Type 'ICanvas | null' is not assignable to type 'ICanvas'.`

Maybe going back to the original if statement type guard would work?

Reply:
Seems that the problem is caused by `private`, change it to `protected` may help.

```diff
- private _canvas: ICanvas | null;
+ protected _canvas: ICanvas | null;
```

Reply:
Still the same issue for me when set as protected.

Reply:
Did you change `canvas` in `asserts` into `_canvas`?